msdk-featdet/msdk-featdet-openms/src/main/java/io/github/msdk/featdet/openms/FeatureFinderMetaboMZValueDetector.java

```diff
@@ -0,0 +1,83 @@
+package io.github.msdk.featdet.openms;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Node;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+
+public class FeatureFinderMetaboMZValueDetector {
+	private final String OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME = "FeatureFinderMetabo";
```

Comment:
This is ok, but it assumes that the FeatureFinderMetabo program is in PATH. Often, that is not the case. It would be nice to add a bit more intelligence to this. Perhaps you can create a new class FeatureFinderMetaboLocator for that purpose. The class can have a single function findFeatureFinderMetabo(), which will contain the logic. The logic can then search some common paths based on platform (/Applications/OpenMS*/TOPP on Mac, C:\Program Files\... on Windows, /usr/local/openMS*/... on Linux). Of course we cannot know for sure where OpenMS is installed, but we try as many possibilities as we can.